### PR TITLE
Added recipe for MedPy

### PR DESCRIPTION
**Pull request by sarthakpati**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sarthakpati** [comment]
Relevant discussion: https://github.com/orgs/conda-forge/teams/bioconda-recipes/discussions/11

**Comment by wolfv** [comment]
Maybe you should also have this discussion on gitter? https://gitter.im/bioconda/Lobby

**Comment by sarthakpati** [comment]
@conda-forge/help-python this is ready for review. I have had a discussion with bioconda folks (see screenshot below) to remove medpy from bioconda once this is merged.

image

**Comment by dopplershift** [comment]
It looks like the bioconda recipe depends on compilers and boost, nominally for the `maxflow` subpackage. Is there any reason this recipe isn't compiling it? (Or starting from the dependency/build from bioconda?)

**Comment by sarthakpati** [comment]
I am not sure. I didn't really have any issues when directly using the pip package for this.

> (Or starting from the dependency/build from bioconda?)

I am not completely sure how to do this. Could you please clarify?

**Comment by dopplershift** [comment]
You can see the old recipe in the [PR you linked](https://github.com/bioconda/bioconda-recipes/pull/28229/files#diff-94eab7630822b18ed44bb34990f422525c004d19b5c1e5462efeabe4f6fad328).

If you want to see if this recipe works, try adding this to `imports` under `test`:
```yml
     - medpy.features
     - medpy.filter
     - medpy.graphcut
     - medpy.io
     - medpy.metric
     - medpy.utilities
```

(taken from the bioconda recipe)

**Comment by dopplershift** [comment]
It looks like `medpy` [optionally compiles](https://github.com/loli/medpy/blob/master/setup.py) an extension module--and this current recipe doesn't do that. This matters because we're replacing the bioconda recipe with this one essentially, and users of the bioconda version might be relying on functionality that needs the `graphcut` module/subpackage.

**Comment by sarthakpati** [comment]
@dopplershift: I got the Linux and osx builds working, but Windows is failing with a very weird error [[ref](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=309879&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=23010c01-6a9d-5f94-9037-792618644b77&l=1234)]:

```powershell
Successfully installed MedPy-0.4.0
Removed build tracker: 'C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\pip-req-tracker-psk5je__'
Traceback (most recent call last):
  File ".ci_support\build_all.py", line 175, in <module>
    build_all(os.path.join(root_dir, "recipes"), args.arch)
  File ".ci_support\build_all.py", line 100, in build_all
    build_folders(recipes_dir, folders, arch, channel_urls)
  File ".ci_support\build_all.py", line 156, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 195, in build
    variants=variants
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 3073, in build_tree
    notest=notest,
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 2159, in build
    windows.build(m, build_file, stats=build_stats, provision_only=provision_only)
  File "C:\Miniconda\lib\site-packages\conda_build\windows.py", line 299, in build
    check_call_env(cmd, cwd=m.config.work_dir, stats=stats, rewrite_stdout_env=rewrite_env)
  File "C:\Miniconda\lib\site-packages\conda_build\utils.py", line 412, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "C:\Miniconda\lib\site-packages\conda_build\utils.py", line 392, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 1.
```

I have basically done a 1:1 copy of the bioconda recipe, so unsure what's wrong... Is this enough to merge?

**Comment by sarthakpati** [comment]
Hi @dopplershift, are you still working on the Windows build for this?

**Comment by sarthakpati** [comment]
I am confused by the error message [[ref](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=340354&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=23010c01-6a9d-5f94-9037-792618644b77&l=1163)]:

```
The syntax of the command is incorrect.
Successfully built MedPy
Installing collected packages: MedPy

Successfully installed MedPy-0.4.0
```

**Comment by bollwyvl** [comment]
Perhaps try doing [`python setup.py build_ext`](https://github.com/search?l=YAML&q=org%3Aconda-forge+build_ext&type=Code) (it might need some flags, etc) before installing... this _might_ give more useful feedback.

**Comment by bollwyvl** [comment]
Heck, when does one of my suggestions work the first time :rofl:

**Comment by sarthakpati** [comment]
Thanks a ton, @bollwyvl!!!

**Comment by sarthakpati** [comment]
@conda-forge/help-python this is ready for review! Thank you.

File: recipes/medpy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+build:
+  number: 0
+  skip: True  # [py2k]
+  script:
+    - export LDFLAGS="${LDFLAGS} -L${PREFIX}/lib"
+    - export CXXFLAGS="${CXXFLAGS} -I${PREFIX}/include -I${PYTHON_INCLUDE_DIR}"
+    - "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv "
+    - python setup.py build_ext --include-dirs=%PREFIX%\include --library-dirs=%PREFIX%\lib  # [win]
+    - python setup.py build_ext --include-dirs=$PREFIX/include --library-dirs=$PREFIX/lib    # [not win]
```
Comment (dopplershift) [inline recipes/medpy/meta.yaml:24]: Wouldn't it make more sense to execute these lines *before* running `pip install`?

Comment (sarthakpati) [reply recipes/medpy/meta.yaml:23]: Sure, changed. I just followed one of the recipes in the link posted by @bollwyvl. Made this change, resolving. 😄

**Comment by sarthakpati** [comment]
Hi @dopplershift, this is ready for merge, now.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**